android-sdk/src/main/java/ai/evolv/android_sdk/EvolvContextImpl.java

```diff
@@ -36,17 +52,75 @@ private void initialize(String uid,
         }
 
         this.uid = uid;
-        this.sid = sid;
-        Class genericRemoteContextClass = remoteContext.getClass();
-        Class genericLocalContextClass = remoteContext.getClass();
-        this.remoteContext = remoteContext != null ? helper
-                .deepCopy(remoteContext,genericRemoteContextClass) : new HashMap<>();
-        this.localContext = localContext != null ? helper
-                .deepCopy(localContext,genericLocalContextClass) : new HashMap<>();
+        if (remoteContext != null) {
+            Class genericRemoteContextClass = remoteContext.getClass();
+            // TODO: 01.06.2021 redo
+            //this.remoteContext = helper.deepCopy(remoteContext, genericRemoteContextClass);
+        } else {
+            //this.remoteContext = new HashMap<>();
+        }
+        if (localContext != null) {
+            Class genericLocalContextClass = localContext.getClass();
+            // TODO: 01.06.2021 redo
+            //this.localContext = helper.deepCopy(localContext, genericLocalContextClass);
+        } else {
+            //this.localContext = new HashMap<>();
+        }
+
         initialized = true;
 
-        // TODO: 25.05.2021 emit
-        //emit(this, CONTEXT_INITIALIZED, this.resolve());
+        Object resolve = resolve();
+        List<Object> objects = new ArrayList<>();
+        objects.add(CONTEXT_INITIALIZED);
```

Comment:
Does this need to be added to the object given it's already the 2nd argument in the emit?

Reply:
in Java, the method signature is strictly defined.
that is, when we need to pass a different number of parameters to one method, we either make a list or use the @overload annotation
![photo_2021-06-03_11-29-27_1](https://user-images.githubusercontent.com/28364912/120613570-092fd900-c45f-11eb-9116-76476b7037f5.jpg)
![photo_2021-06-03_11-29-19_2](https://user-images.githubusercontent.com/28364912/120613584-0b923300-c45f-11eb-8675-4de4e32e2c64.jpg)